plugin/storage/es/spanstore/reader_test.go

```diff
@@ -307,6 +327,8 @@ func TestSpanReader_GetTrace(t *testing.T) {
 			}, nil)
 
 		trace, err := r.reader.GetTrace(context.Background(), model.NewTraceID(0, 1))
+		require.NotEmpty(t, r.exporter.GetSpans(), "Spans recorded")
+		require.NoError(t, r.tracerCloser())
```

Comment:
one of the nice thing about doing tests in a functional style via `withSpanReader` is that `withSpanReader` can do both initialization and tear-down, since it's frame is lower in the stack trace than the actual test function and `defer`s inside `withSpanReader` always get executed even if the test function panics. 
```
   TestSomething()
       withSpanReader(testFn)
            d = initialize()
            defer d.teardown()
            testFn(d)
```

So I don't think we even need to expose traceCloser in the struct, we can just defer-call it inside `withSpanReader`.

This is very different from helper functions that simply _return_ a tracer. They have no room to defer-schedule shutdown, and you have to do it manually in the caller.

Reply:
That's nit. Cherry picked. 😄